### PR TITLE
auto-documentation

### PR DESCRIPTION
**Pull request by dima618**
Addeda auto generated docs to backend. Details are available in the README file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,8 +72,16 @@ $ npm run test:e2e
...
+More detailed information on using compodoc can be found [here](https://compodoc.app/guides/getting-started.html).
+```bash
+# generate documentation
+$ npx @compodoc/compodoc -p tsconfig.json -s
+``` 
+Documentation will be generated at http://localhost:8080
+
 ## Docker Notes
-### persistence
+### Persistence
```
Comment (cubscout61) [inline README.md:84]: Thanks for the capatalization

**Comment by 121Eagle** [comment]
For the final commit I have to say that that has no problems with me.

File: src/roles/roles.guard.ts [hunk 2]
```diff
@@ -10,6 +10,11 @@ import { ROLES_KEY } from './roles.decorator';
 export class RolesGuard implements CanActivate {
     constructor (private reflector: Reflector) {}
 
+    /**
+     * This method verifies the user role.
+     * @param context Execution context
+     * @returns True if user role matches the restriction, false otherwise.
+     */
```
Comment (121Eagle) [inline src/roles/roles.guard.ts]: Yep, I have no issues with adding in comments to the code

**Comment by 121Eagle** [comment]
Yeah I feel that this is perfectly fine.

**Review by 121Eagle — APPROVED** [review]
I agree with all of these changes

**Pull request merged**